src/fidesctl/core/system.py

```diff
@@ -154,25 +154,51 @@ def generate_system_aws(
     return file_name
 
 
-def generate_system_okta(
-    okta_config: Optional[OktaConfig], file_name: str, include_null: bool
-) -> str:
+async def generate_okta_systems(
+    organization: Organization, okta_config: Optional[OktaConfig]
+) -> List[System]:
     """
-    Given an organization url, generates a system manifest from existing Okta
-    applications.
+    Given an okta configuration, calls okta for a list of
+    applications and returns the corresponding systems.
     """
 
-    _check_okta_connector_import()
-
     import fidesctl.connectors.okta as okta_connector
 
     okta_client = okta_connector.get_okta_client(okta_config)
-    okta_applications = asyncio.run(
-        okta_connector.list_okta_applications(okta_client=okta_client)
+    okta_applications = await okta_connector.list_okta_applications(
+        okta_client=okta_client
     )
     okta_systems = okta_connector.create_okta_systems(
-        okta_applications=okta_applications
+        okta_applications=okta_applications, organization_key=organization.fides_key
     )
+    return okta_systems
+
+
+def generate_system_okta(
```

Comment:
This is actually git trickery as `generate_system_okta` is still there. That is a function which executes the `generate system okta` command. I would like to actually move the `generate_system_okta` and `generate_system_aws` functions out of `system.py` since I think they belong with the cli code in `generate.py`. 

`generate_okta_systems` is the function i'd like to keep in `system.py` and it should be called from the cli and api code to do as it says.